crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -108,17 +116,62 @@ impl Converter for CumulocityConverter {
     }
 
     fn try_init_messages(&self) -> Result<Vec<Message>, ConversionError> {
-        let ops = Operations::try_new("/etc/tedge/operations", "c8y")?;
-        let ops = ops.get_operations_list();
-        let ops = ops.iter().map(|op| op as &str).collect::<Vec<&str>>();
-
-        let ops_msg = SmartRestSetSupportedOperations::new(&ops);
-        let topic = Topic::new_unchecked("c8y/s/us");
-        let msg = Message::new(&topic, ops_msg.to_smartrest()?);
-        Ok(vec![msg])
+        let fragments_message = create_inventory_fragments_message(&self.device_name)?;
+
+        let supported_operations_message = create_supported_operations_fragments()?;
+
+        Ok(vec![supported_operations_message, fragments_message])
     }
 }
 
+fn create_supported_operations_fragments() -> Result<Message, ConversionError> {
+    let ops = Operations::try_new("/etc/tedge/operations")?;
+    let ops = ops.get_operations_list("c8y");
+
+    let ops_msg = SmartRestSetSupportedOperations::new(&ops);
+    let topic = Topic::new_unchecked("c8y/s/us");
+    Ok(Message::new(&topic, ops_msg.to_smartrest()?))
+}
+
+fn create_inventory_fragments_message(device_name: &str) -> Result<Message, ConversionError> {
+    let ops_msg = get_inventory_fragments(INVENTORY_FRAGMENTS_FILE_LOCATION)?;
+
+    let topic = Topic::new_unchecked(&format!(
+        "c8y/inventory/managedObjects/update/{}",
+        device_name
+    ));
+    Ok(Message::new(&topic, ops_msg.to_string()))
+}
+
+/// reads a json file to serde_json::Value
+///
+/// # Example
+/// ```
+/// let json_value = read_json_from_file("/path/to/a/file").unwrap();
+/// ```
+fn read_json_from_file(file_path: &str) -> Result<serde_json::Value, ConversionError> {
+    let mut file = File::open(Path::new(file_path))?;
+    let mut data = String::new();
+    file.read_to_string(&mut data)?;
+    let json: serde_json::Value = serde_json::from_str(&data)?;
+    Ok(json)
+}
+
+/// gets a serde_json::Value of inventory
+fn get_inventory_fragments(file_path: &str) -> Result<serde_json::Value, ConversionError> {
+    let agent_fragment = C8yAgentFragment::new();
+    let json_fragment = agent_fragment.to_json()?;
+
+    match read_json_from_file(file_path) {
+        Ok(mut json) => {
+            json.as_object_mut()
+                .ok_or_else(|| return ConversionError::FromOptionToResultConversion)?
+                .insert("c8y_Agent".to_string(), json_fragment);
+            Ok(json)
+        }
+        Err(_) => Ok(json_fragment),
```

Comment:
It may be good to be a little more explicit in the error type and maybe let the user know that the json is incorrect or that the file doesn't exist.